lib/common/ipc_attrd.c

```diff
@@ -87,11 +87,53 @@ create_attrd_op(const char *user_name)
 
     crm_xml_add(attrd_op, F_TYPE, T_ATTRD);
     crm_xml_add(attrd_op, F_ORIG, (crm_system_name? crm_system_name: "unknown"));
-    crm_xml_add(attrd_op, PCMK__XA_ATTR_USER, user_name);
+
+    if (user_name != NULL) {
+        crm_xml_add(attrd_op, PCMK__XA_ATTR_USER, user_name);
+    }
 
     return attrd_op;
 }
 
+static int
+send_attrd_request(pcmk_ipc_api_t *api, xmlNode *request)
+{
+    /* FIXME: What to do about the fact that query wants a reply? */
+    pcmk__send_ipc_request(api, request);
+    return pcmk_rc_ok;
```

Comment:
I was going to revisit this function after figuring out what to do with the FIXME.  It doesn't feel like I need to keep track of queries, so I can just have this function return the result of pcmk__send_ipc_request.  Really, I don't even need send_attrd_request if that's all it does but I'd like to keep it in case we need to do more here later.